test.py

```diff
@@ -381,6 +381,75 @@ def test_restore(self):
         with open(join(path, 'foobar')) as f:
             self.assertEqual(f.read(), 'modified foobar')
 
+    def test_purge(self):
+        """Check we can purge snapshots with a save pattern
+        """
+        name = PREFIX_TEST_VOLUME + uuid.uuid4().hex
+        # first run the purge without snapshots (should do nothing)
+        resp = self.app.post('/VolumeDriver.Snapshots.Purge',
+                             json.dumps({'Name': name, 'Pattern': '2h:2h'}))
+        path = join(VOLUMES_PATH, name)
+        self.app.post('/VolumeDriver.Create', json.dumps({'Name': name}))
+        with open(join(path, 'foobar'), 'w') as f:
+            f.write('original foobar')
+
+        def cleanup_snapshots():
+            for s in os.listdir(SNAPSHOTS_PATH):
+                if s.startswith(PREFIX_TEST_VOLUME):
+                    btrfs.Subvolume(join(SNAPSHOTS_PATH, s)).delete()
```

Comment:
as far I remember, I've a bit change that behaviour, you could use btrfs.Subvolume(join(SNAPSHOTS_PATH, PREFIX_TEST_VOLUME) + '*').delete()